docs/authentication.md

```diff
@@ -0,0 +1,77 @@
+# Authentication
+
+This document details how to configure your Helios installation to authenticate
+requests between the Helios CLI and the Helios masters/cluster.
+
+# TODOs for this document
+
+- configuration needed to setup master to require authentication
+  - make sure authentication can be enabled based on client version header, to
+    apply to versions >= some value and/or completely
+- add notion of schemes
+- link to how to setup https for helios masters
```

Comment:
I had docs in this PR: https://github.com/spotify/helios/pull/655/files#diff-ca6bc2e6c7f2fbcadd6202b5707a25adR108